### PR TITLE
enrollment add activate, deactivate and delete functionality.

### PR DESCRIPTION
**Pull request by Aciid**
Add functionality to activate, deactivate and delete invite-codes

### CODE REVIEW

**Comment by Aciid** [comment]
Workflow:

Added activate column to management table, not checked anywhere yet.

## Create invite-code for admin user_mnagement_hash, one from init or one that has been created.
```
curl --location 'http://localhost:8000/api/v1/enrollment/invitecode/create' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--data '{
  "user_management_hash": "TestikalukalukalukaulinJotainAsdJotainJotainJotain"
}'
```
Returns:
```
{
  "invite_code": "AP79VI9Q"
}
```

## Check validiy for known invite-code
```
curl --location 'http://localhost:8000/api/v1/enrollment/invitecode?invitecode=AP79VI9Q' \
--header 'Accept: application/json'
```
```
{
  "invitecode_is_active": true
}
```

## Activate known-invite-code ( modify active column to 1 ) 
```
curl --location --request PUT 'http://localhost:8000/api/v1/enrollment/invitecode/activate' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--data '{
  "invite_code": "AP79VI9Q"
}'
```
```
{
  "invite_code": "AP79VI9Q"
}
```


## Deactivate known invite-code ( modify active column to 0
```
curl --location --request PUT 'http://localhost:8000/api/v1/enrollment/invitecode/deactivate' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--data '{
  "invite_code": "AP79VI9Q"
}'
```
```
{
  "invite_code": "AP79VI9Q"
}
```

## Delete known invite-code
`curl --location --request DELETE 'http://localhost:8000/api/v1/enrollment/invitecode/AP79VI9Q'`

```
{
  "invite_code": "AP79VI9Q"
}
```

File: src/rasenmaeher_api/web/api/enrollment/schema.py [hunk 1]
```diff
@@ -620,10 +620,52 @@ class EnrollmentInviteCodeCreateOut(BaseModel):  # pylint: disable=too-few-publi
     invite_code: str
 
 
+class EnrollmentInviteCodeDeactivateIn(BaseModel):
```
Comment (rambo) [inline src/rasenmaeher_api/web/api/enrollment/schema.py:623]: Use extra="forbid", per https://github.com/pvarki/python-rasenmaeher-api/issues/33

File: src/rasenmaeher_api/web/api/enrollment/views.py [hunk 2]
```diff
@@ -1,8 +1,10 @@
 """Enrollment API views."""
+from sre_constants import SUCCESS
```
Comment (rambo) [inline src/rasenmaeher_api/web/api/enrollment/views.py]: Group imports, stdlib ones on top (preferably starting with "from typing" -imports), then empty line then 3rd party library imports then empty line and package internal imports (using relative imports)

File: src/rasenmaeher_api/web/api/enrollment/views.py [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
 import string
 import random
 import logging
-from typing import Dict, List, Any, Optional
+from typing import Dict, List, Any, Optional, Union
+from unittest import result
```
Comment (rambo) [inline src/rasenmaeher_api/web/api/enrollment/views.py]: What is this import doing in a view, also we use pytest...

File: src/rasenmaeher_api/web/api/enrollment/views.py [hunk 4]
```diff
@@ -785,7 +841,7 @@ async def post_invite_code(
 
     # Random string for invite-code eg. GLXBT0
     _invite_code = "".join(random.choices(string.ascii_uppercase + string.digits, k=8))  # nosec B311
-
+    print(_invite_code)
```
Comment (rambo) [inline src/rasenmaeher_api/web/api/enrollment/views.py]: Never print, always use LOGGER.debug check https://github.com/pvarki/python-rasenmaeher-api/blob/main/src/rasenmaeher_api/jwtinit.py for reminder on standard LOGGER init

**Review by rambo — CHANGES_REQUESTED** [review]

File: src/rasenmaeher_api/web/api/enrollment/views.py (hunk 4 above)
Comment (Aciid) [reply src/rasenmaeher_api/web/api/enrollment/views.py]: 👍

**Comment by Aciid** [comment]
API tests failing because they are skeleton.

**Review by rambo — APPROVED** [review]
We'll fix the test weirdness later

**Pull request merged**